org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/window/WRecordIDDialog.java

```diff
@@ -117,12 +124,22 @@ private void init(Page page) {
 		contentDiv = new Div();
 		confirmPanelDiv = new Div();
     	okBtn = ButtonFactory.createNamedButton(ConfirmPanel.A_OK, true, true);
-    	cancelBtn = ButtonFactory.createNamedButton(ConfirmPanel.A_CANCEL, true, true);;
-    	tableIDEditor = new WTableDirEditor("AD_Table_ID", false, false, true, tableIDGridField.getLookup(), true);
+    	cancelBtn = ButtonFactory.createNamedButton(ConfirmPanel.A_CANCEL, true, true);
+    	
+    	if(DisplayType.Search == tableIDGridField.getDisplayType()) {
+    		tableIDEditor = new WSearchEditor("AD_Table_ID", false, false, true, tableIDGridField.getLookup());
+    	}
+    	else {
+    		tableIDEditor = new WTableDirEditor("AD_Table_ID", false, false, true, tableIDGridField.getLookup(), true);
+    	}
+    	tableIDEditor.addValueChangeListener(this);
 		tableIDEditor.setValue(tableIDValue);
+		
 		int tableID = tableIDValue != null ? tableIDValue.intValue() : 0;
-		recordsEditor = tableID > 0 ? new WSearchEditor("Record_ID", false, false, true, editor.getRecordsLookup(tableID)) : null;
-    	
+		MLookup recordsLookup = editor.getRecordsLookup(tableID);
+		if(recordsLookup != null)
+			recordsEditor = tableID > 0 ? new WSearchEditor("Record_ID", false, false, true, recordsLookup) : null;
```

Comment:
you are doing duplicate checking here, if recordsLookup is not null, tableID must > 0.